src/adapters/book-adapters.ts

```diff
@@ -0,0 +1,254 @@
+import { setCookie, getCookie } from "../util/Authentication"
+export const fetchAPIData = async (url: string, method: string, body:object | undefined , headers?:object) => {
+    const cookie = getCookie('shelf-share-session')as { token: string }
+    const token = cookie?.token
+    const response = await fetch(url , {
+        method,
+        body: JSON.stringify(body),
+        headers: {
+            Authorization: `Bearer ${token}`,
+            'Content-Type': 'application/json',
+            ...headers,
+        },
+    })
+    
+    if(response.status.toString().startsWith('4')||response.status.toString().startsWith('5')){
+      console.log("Error", response.status)
+    }
+    else{
+      const data = await response.json();  
+      return data; 
+    }
+        
+}
+// Create book adapter
+type bookInput = {
+    title: string,
+    language: string,
+    ageRange: string,
+    publishingYear: number,
+    status: string,
+    image?: Buffer,
+    description: string,
+    genre: string,
+    author:string
+}
+/**
+ * This function will create a book with the provided information.
+ * @param {Object} bookInput - The input object containing information for the book creation.
+ * @param {String} bookInput.title - Book's title.
+ * @param {String} bookInput.language - Book's language.
+ * @param {String} bookInput.ageRange - Book's age-range.
+ * @param {Number} bookInput.publishingYear - Book's publishing year.
+ * @param {String} bookInput.status - Book's status.
+ * @param {Buffer} bookInput.image - Book's image(optional).
+ * @param {String} bookInput.description - Book's description.
+ * @param {String} bookInput.genre - Book's genre.
+ * @param {String} bookInput.author - Book's author.
+ * @example
+ * const bookInput = {
+ *  title = "Pachinko",
+ *  language = "English",
+ *  ageRange = "Adults",
+ *  publishingYear = 2017,
+ *  status = "Open",
+ *  description = "Pachinko is the second novel by Harlem-based author and journalist Min Jin Lee.
+ *  Published in 2017, Pachinko is an epic historical fiction novel following a Korean family who immigrates to Japan.",
+ *  genre = "Historical Fiction",
+ *  author = "Min Jin Lee"
+ * };
+ * export const createBookAdapter = async (bookInput) => {
+ *  // function implementation here
+ * };
```

Comment:
This can just say createBookAdapter(bookInput) as a sufficient example. Can you change this for the other examples as well?

Reply:
Do you mean const bookInput needs to be changed to createBookAdapter(bookInput)? Other than that part in this highlight section stay the same?